src/commands/partial visibility question/anonymous.ts

```diff
@@ -0,0 +1,99 @@
+import { ApplicationCommandOptionData, CommandInteraction, Message, MessageEmbed, TextChannel } from 'discord.js';
+import { generateQuestionId } from '@lib/utils';
+import { Course } from '@lib/types/Course';
+import { SageUser } from '@lib/types/SageUser';
+import { PVQuestion } from '@lib/types/PVQuestion';
+import { BOT, DB, MAINTAINERS } from '@root/config';
+import { Command } from '@lib/types/Command';
+
+export default class extends Command {
+
+	description = 'Send an anonymous question in your classes general channel.';
+	usage = '[course] <question>';
+	extendedHelp = `${BOT.NAME} will automatically determine your course if you are only enrolled in one!`;
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'question',
+			description: 'What would you like to ask?',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'course',
+			description: 'What course chat would you like to ask your question in?',
+			type: 'STRING',
+			required: false
+		}
+	]
+
+	run(_msg: Message): Promise<void> { return; }
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		const user: SageUser = await interaction.client.mongo.collection(DB.USERS).findOne({ discordId: interaction.user.id });
+
+		if (!user) {
+			const responseEmbed = new MessageEmbed()
+				.setTitle(`Error`)
+				.setDescription(`Something went wrong. Please contact ${MAINTAINERS}`)
+				.setColor('#ff0000');
+			return interaction.reply({ embeds: [responseEmbed], ephemeral: true });
+		}
+
+		let course: Course;
+		const question = interaction.options.getString('question');
+		const courses: Array<Course> = await interaction.client.mongo.collection(DB.COURSES).find().toArray();
+
+		if (user.courses.length === 1) {
+			course = courses.find(c => c.name === user.courses[0]);
+		} else {
+			const inputtedCourse = courses.find(c => c.name === interaction.options.getString('course'));
+			if (!inputtedCourse) {
+				const responseEmbed = new MessageEmbed()
+					.setTitle(`Argument error`)
+					.setDescription('I wasn\'t able to determine your course based off of your enrollment or your input. Please specify the course at the beginning of your question.' +
```

Comment:
Shouldn't it be specified using the argument?

Reply:
Unless I don't quite get what you're saying, this error message comes up if the user is enrolled in less/more than 1 class and does not specify the class they'd like to ask their question in (since the parameter is optional since Sage auto-detects your class if you're enrolled in just one).

If I've misinterpreted this comment, please correct me